07_errors/n0npax/errors.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+)
+
+// Error codes
+const (
+	ErrInvalidInputCode = 422
+	ErrNotFoundCode     = 404
+)
+
+// Error wrapps errors with code, message and error itself
+type Error struct {
+	Message string
+	Code    int
+}
+
+func (e *Error) Error() string {
+	return fmt.Sprintf("Error code: %v, message : %v", e.Code, e.Message)
+}
+
+// Errorf creates a new Error with formatting
+func Errorf(code int, msg string) *Error {
+	myError := &Error{
+		Message: msg,
+		Code:    code,
+	}
+	return myError
```

Comment:
The `Error` struct is small enough you don't need to use named fields. I'd make this a one-liner:

```
return &Error{msg, code}
```

To me, that's more readable because it encapsulates just the essential components and nothing else.

Reply:
fixed